keps/sig-auth/541-external-credential-providers/README.md

```diff
@@ -493,6 +494,102 @@ credentials).
 PEM format. The certificate must be valid at the time of execution. These
 credentials are used for mTLS handshakes.
 
+### Metrics
+
+As discussed [below](#rollout-upgrade-and-rollback-planning), there are 3
+primary metrics used by this feature set.
+
+```golang
+var (
+  execPluginCertTTL = k8smetrics.NewGaugeFunc(
+    k8smetrics.GaugeOpts{
+      Name: "rest_client_exec_plugin_ttl_seconds",
+      Help: "Gauge of the shortest TTL (time-to-live) of the client " +
+        "certificate(s) managed by the auth exec plugin. The value " +
+        "is in seconds until certificate expiry (negative if " +
+        "already expired). If auth exec plugins are unused or manage no " +
+        "TLS certificates, the value will be +INF.",
+    },
+    func() float64 {
+      if execPluginCertTTLAdapter.e == nil {
+        return math.Inf(1)
+      }
+      return execPluginCertTTLAdapter.e.Sub(time.Now()).Seconds()
+    },
+  )
+
+  execPluginCertRotation = k8smetrics.NewHistogram(
+    &k8smetrics.HistogramOpts{
+      Name: "rest_client_exec_plugin_certificate_rotation_age",
```

Comment:
This metric is very oddly named. I prefer `rest_client_exec_plugin_certificate_lifespan_duration` or something.

Reply:
Yeah, I agree it is a little weird.

What is our API contract with respect to metric names? I _think_ this metric has already been running in clusters since `v1.18.0`: https://github.com/kubernetes/kubernetes/pull/84382/commits/e3b1cd14c31375e52e267ce7e4246f80ee26ddba. Are we allowed to change metric names between releases?

Reply:
Looks like it came in in https://github.com/kubernetes/kubernetes/pull/84382#pullrequestreview-313078260

I'd prefer we keep the current name unless it is broken.